subsys/net/lib/lwm2m/lwm2m_obj_firmware.c

```diff
@@ -348,31 +379,36 @@ static int lwm2m_firmware_init(const struct device *dev)
 	int ret = 0;
 
 	/* Set default values */
-	package_uri[0] = '\0';
-	/* Initialize state machine */
-	/* TODO: should be restored from the permanent storage */
-	update_state = STATE_IDLE;
-	update_result = RESULT_DEFAULT;
-#ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
-	delivery_method = DELIVERY_METHOD_BOTH;
-#else
-	delivery_method = DELIVERY_METHOD_PUSH_ONLY;
-#endif
-
 	firmware.obj_id = LWM2M_OBJECT_FIRMWARE_ID;
 	firmware.version_major = FIRMWARE_VERSION_MAJOR;
 	firmware.version_minor = FIRMWARE_VERSION_MINOR;
 	firmware.is_core = true;
 	firmware.fields = fields;
 	firmware.field_count = ARRAY_SIZE(fields);
-	firmware.max_instance_count = 1U;
+	firmware.max_instance_count = MAX_INSTANCE_COUNT;
 	firmware.create_cb = firmware_create;
 	lwm2m_register_obj(&firmware);
 
-	/* auto create the only instance */
-	ret = lwm2m_create_obj_inst(LWM2M_OBJECT_FIRMWARE_ID, 0, &obj_inst);
-	if (ret < 0) {
-		LOG_DBG("Create LWM2M instance 0 error: %d", ret);
+
+	for (int idx = 0; idx < MAX_INSTANCE_COUNT; idx++) {
+		package_uri[idx][0] = '\0';
+
+		/* Initialize state machine */
+		/* TODO: should be restored from the permanent storage */
+		update_state[idx] = STATE_IDLE;
+		update_result[idx] = RESULT_DEFAULT;
+#ifdef CONFIG_LWM2M_FIRMWARE_UPDATE_PULL_SUPPORT
+		delivery_method[idx] = DELIVERY_METHOD_BOTH;
+#else
+		delivery_method[idx] = DELIVERY_METHOD_PUSH_ONLY;
+#endif
+		LOG_INF("Creating LWM2M FW instance %d", idx);
```

Comment:
I think this is too verbose. Either DBG or drop the whole line. Previously it did not print anything for a success case. And I would also prefer not to print, as this can rarely fail, and is part of so normal initialization flow that it should be obvious.